docs(js): Enhance initialScope options docs (#3756)
getsentry/sentry-docs: src/platforms/common/configuration/options.mdx

<ConfigKey name="initial-scope" supported={["javascript","node"]}>

> Data to be set to the initial scope. Initial scope can be defined either as an object as follows

Reviewer: This construction threw me off for a second because I hadn't read further down the page yet to find the "or" so I just assumed it was missing in the first statement. Suggest making this a bit more foolproof.

```suggestion
Data to be set to the initial scope. Initial scope can be defined either as an object or a callback function, as shown below.

Object:
```

```javascript
Sentry.init({
  dsn: "___PUBLIC_DSN___",
  debug: true,

>   initialScope: {
>     scope.setTag("my-tag", "my value");
>     scope.setUser({id: 42, email: "john.doe@example.com"});
>   }

Reviewer 2: This is not a valid code sample, you can't call `scope.setTag` it should be an object.
```suggestion
  initialScope: {
    tags: {"my-tag": "my value"},
    user: { id: 42, email: "john.doe@example.com"},
  }
```

Author: Oops xd

});
```

> Or as a callback function

Reviewer: ```suggestion
Callback function
```

<ConfigKey name="initial-scope" supported={["javascript","node"]}>

Data to be set to the initial scope. Initial scope can be defined either as an object or a callback function, as shown below.

Object:

```javascript
Sentry.init({
  dsn: "___PUBLIC_DSN___",
  debug: true,
  initialScope: {
    tags: {"my-tag": "my value"},
    user: {id: 42, email: "john.doe@example.com"},
  }
});
```

> Callback function

Author: ```suggestion
Callback function:
```
